README.md

```diff
@@ -79,52 +83,70 @@ substoich =[2 0 0;
-                    prodstoich::AbstractMatrix; 
-                    species::AbstractVector=Operation[], 
-                    params::AbstractVector=Operation[])
+prn = loadrxnetwork(mn::MatrixNetwork) #or
+prn = loadrxnetwork(cmn::ComplexMatrixNetwork)
 ```
-Here `MatrixNetwork()` is the dispatch type, which selects that we are
-constructing a matrix-based stoichiometric representation as input. The other
-parameters are:
+Here `MatrixNetwork` and `ComplexMatrixNetwork` are the types, which selects that we are
+constructing a substrate/products stoichiometric matrix-based or complex- matrix based stoichiometric representation as input. These types have 
+following fields :
 - `rateexprs` - Any valid `ModelingToolkit.Operation`s for the rates, or basic
   number types. This can be a hardcoded rate constant like `1.0`, a a parameter
   like `k1` above, or an `Operation` involving parameters and species like
   `k*A`. Note, the reaction `A+B --> C` with rate `k*B` would have rate law
   `k*A*B^2`.
-- `substoich` - A number of species by number of reactions matrix with entry
-  `(i,j)` giving the stoichiometric coefficient of species `i` as a substrate in
-  reaction `j`. 
-- `prodstoich` - A number of species by number of reactions matrix with entry
-  `(i,j)` giving the stoichiometric coefficient of species `i` as a product in
-  reaction `j`.
+- matrix inputs 
+  - For `MatrixNetwork`
+    - `substoich` - A number of species by number of reactions matrix with entry
+  ` (i,j)` giving the stoichiometric coefficient of species `i` as a substrate in
+    reaction `j`. 
+    - `prodstoich` - A number of species by number of reactions matrix with entry
+  `  (i,j)` giving the stoichiometric coefficient of species `i` as a product in
+     reaction `j`.
+  - For `ComplexMatrixNetwork`
+  
+    - `stoichmat` - Complex stoichiometrc matrix , [defined here](https://catalyst.sciml.ai/dev/api/catalyst_api/#Catalyst.complexstoichmat).
+    - `incidencemat` - Complex incidence matrix, [defined here](https://catalyst.sciml.ai/dev/api/catalyst_api/#Catalyst.reactioncomplexes)
+ 
 - `species` - Optional `ModelingToolkit.Operation`s representing each species in
   the network.
 - `parameters` - Optional `ModelingToolkit.Operation`s representing each
   parameter in the network.
+- `t` - Optional `ModelingToolkit.Operation`s representing time as independent variable of the reaction network
 
 `prn` is again a `ParsedReactionNetwork`, with only the `reaction_network`
 field, `prn.rn`, defined.
 
-A dispatch is added if `substoich` and `prodstoich` both have the type
-`SparseMatrixCSC`, in which case they are efficiently iterated through using the
+Dispatches are added if `substoich` and `prodstoich` both have the type
+`SparseMatrixCSC`in case of `MatrixNetwork`(or `stoichmat` and `incidencemat` both as `SparseMatrixCSC` type in case of `ComplexMatrixNetwork`), in which case they are efficiently iterated through using the
```

Comment:
or `stoichmat` and `incidencemat` both have the type `SparseMatrixCSC` in case of